### PR TITLE
add rabbit requeue logic

### PR DESCRIPTION
**Pull request by jackschofield23**
Changes required for #93 
also supports #224 

### Description

Adds rabbit re-queue logic to support a queue delay and retry count

### Status
**DONE**

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] All tests passed locally.
- [x] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.
- [ ] User guide updated.
- [ ] I have updated the changelog
- [x] Any new files have copyright headers
- [x] Code coverage above 70%
- [x] TODO items resolved

### CODE REVIEW

File: src/WorkflowManager/Program.cs [hunk 1]
```diff
@@ -112,11 +112,6 @@ private static void Main(string[] args)
         {
             var host = CreateHostBuilder(args).Build();
 
-            if (args.Length == 0)
```
Comment (neildsouth) [inline src/WorkflowManager/Program.cs:115]: This change will mean the TaskManager wont start when the app is run ?

Comment (jackschofield23) [reply src/WorkflowManager/Program.cs:115]: Remember when you spotted something like the task manager running twice? This was the cause of it. It was starting it for a second time. It is already started in the AddTaskManager step.

File: src/TaskManager/TaskManager.cs [hunk 2]
```diff
@@ -96,18 +96,22 @@ private async Task TaskCallbackEventReceivedCallback(MessageReceivedEventArgs ar
...
+            JsonMessage<TaskCallbackEvent>? message = null;
             try
             {
+                message = args.Message.ConvertToJsonMessage<TaskCallbackEvent>();
                 await HandleTaskCallback(message).ConfigureAwait(false);
             }
             catch (OperationCanceledException ex)
             {
                 _logger.ServiceCancelledWithException(ServiceName, ex);
+                await _messageBrokerSubscriberService!.RequeueWithDelay(message);
```
Comment (lillie-dae) [inline src/TaskManager/TaskManager.cs:109]: can ConvertToJsonMessage fail here?

File: src/TaskManager/TaskManager.cs [hunk 3]
```diff
@@ -198,15 +202,18 @@ private async Task HandleTaskCallback(JsonMessage<TaskCallbackEvent> message)
...
                 {
-                    updateMessage.Body.Status = TaskExecutionStatus.Failed;
-                    updateMessage.Body.Reason = FailureReason.PluginError;
-
                     _logger.MetadataRetrievalFailed(ex);
+                    await HandleMessageException(message, message.Body.WorkflowInstanceId, message.Body.TaskId, message.Body.ExecutionId, ex.Message, false).ConfigureAwait(false);
+
+                    return;
                 }
             }
```
Comment (lillie-dae) [inline src/TaskManager/TaskManager.cs:218]: this whole section feels like its adding a lot of complexity to handle task callback which is like 100 lines long, could it be moved to a separate function from seems like its all to do with metadata from that point?

line 184 
`            if (!string.IsNullOrWhiteSpace(metadataAssembly))`

File: src/TaskManager/TaskManager.cs (hunk 2 above)
Comment (jackschofield23) [reply src/TaskManager/TaskManager.cs]: yeah it could, i've pulled this from the args directly now

File: src/TaskManager/TaskManager.cs (hunk 3 above)
Comment (jackschofield23) [reply src/TaskManager/TaskManager.cs:202]: Pulled what i could out into a separate method, looks cleaner now

**Pull request merged**